src/verifast.ml

```diff
@@ -2617,9 +2624,13 @@ module VerifyProgram(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
     | _ -> with_context (Executing (h, env, l, "Freeing parameters.")) (fun _ -> cleanup_heapy_locals_core (pn, ilist) l h env ps cont)
 
   and compute_heapy_params loc func_kind params pre_tenv ss =
-    let penv = get_unique_var_symbs_ params (match func_kind with Regular -> false | _ -> true) in
+    let is_ghost = func_kind |> function Regular -> false | _ -> true in
+    let penv = params |> List.map @@ function 
+      | (x, RefType t) -> x, get_unique_var_symb_ (x ^ "_addr") t is_ghost

```

Comment:
Het type dat je meegeeft aan `get_unique_var_symb` moet hier `PtrType t` zijn, niet `t`? (Nu, ik denk dat het provertype in elk geval toch int gaat zijn, aangezien je geen references naar inductives kunt hebben. Behalve misschien in het geval van references naar bool?)